src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -80,7 +81,15 @@ public FeedbackSessionAttributes getFeedbackSession(String courseId, String feed
 
         if (feedbackSession != null && feedbackSession.isSessionDeleted()) {
             log.info("Trying to access soft-deleted session: " + feedbackSessionName + "/" + courseId);
-            return null;
+        }
+        if (feedbackSession == null || feedbackSession.isSessionDeleted()) {
+            String errorMessage = "";
+            if (feedbackSession == null) {
+                errorMessage = "Session not found.";
+            } else {
+                errorMessage = "Session is deleted.";
+            }
+            throw new EntityNotFoundException(new EntityDoesNotExistException(errorMessage));
```

Comment:
This violates our policy on DB layer though https://github.com/TEAMMATES/teammates/blob/master/docs/design.md#policies-2

Reply:
There are two very dangerous things done here:
- Assuming that feedback session not found is not desirable. DB layer has no concern with such.
- Throwing unchecked exception in DB layer.

Reply:
The problem was that this issues is quite prevalent. There is almost never a null check for `logic.getFeedbackSession()` in the access control of feedback session actions. I've moved the check to `checkSpecificAccessControl`